org/postgresql/geometric/PGline.java

```diff
@@ -17,38 +18,67 @@
 import java.sql.SQLException;
 
 /**
- * This implements a line consisting of two points.
+ * This implements a line represented by the linear equation Ax + By + C = 0
  *
- * Currently line is not yet implemented in the backend, but this class
- * ensures that when it's done were ready for it.
- */
+ **/
 public class PGline extends PGobject implements Serializable, Cloneable
 {
+
+    /**
+     * Coefficient of x
+     */
+    public double a;
+
+    /**
+     * Coefficient of y
+     */
+    public double b;
+
     /**
-     * These are the two points.
+     * Constant
      */
-    public PGpoint point[] = new PGpoint[2];
+    public double c;
 
     /**
-     * @param x1 coordinate for first point
-     * @param y1 coordinate for first point
-     * @param x2 coordinate for second point
-     * @param y2 coordinate for second point
+     * @param a coefficient of x
+     * @param b coefficient of y
+     * @param c constant
+     */
+    public PGline(double a, double b, double c) {
+        this();
+        this.a = a;
+        this.b = b;
+        this.c = c;
+    }
+
+    /**
+     * @param x1 coordinate for first point on the line
+     * @param y1 coordinate for first point on the line
+     * @param x2 coordinate for second point on the line
+     * @param y2 coordinate for second point on the line
      */
     public PGline(double x1, double y1, double x2, double y2)
     {
-        this(new PGpoint(x1, y1), new PGpoint(x2, y2));
+        this();
+        double a = (y2 - y1) / (x2 - x1);
```

Comment:
Does that properly handle "1.0/0.0", "0.0/0.0", "1/NaN", etc cases?

If it does, can you please add relevant comment here?